pkg/gvisor/enable.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gvisor
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/pkg/errors"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+const (
+	nodeDir = "/node"
+)
+
+// Enable follows these steps for enabling gvisor in minikube:
+//   1. creates necessary directories for storing binaries and runsc logs
+//   2. downloads runsc and gvisor-containerd-shim
+//   3. copies necessary containerd config files
+//   4. restarts containerd
+func Enable() error {
+	if err := makeGvisorDirs(); err != nil {
+		return errors.Wrap(err, "creating directories on node")
+	}
+	if err := downloadBinaries(); err != nil {
+		return errors.Wrap(err, "downloading binaries")
+	}
+	if err := copyConfigFiles(); err != nil {
+		return errors.Wrap(err, "copying config files")
+	}
+	if err := restartContainerd(); err != nil {
+		return errors.Wrap(err, "restarting containerd")
+	}
+	// When pod is terminated, disable gvisor and exit
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		if err := Disable(); err != nil {
+			log.Printf("Error disabling gvisor: %v", err)
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}()
+	log.Print("gvisor successfully enabled in cluster")
+	// sleep for one year so the pod continuously runs
+	time.Sleep(24 * 7 * 52 * time.Hour)
```

Comment:
nit: this will make the code block indefinitely:
select {}

Reply:
thanks!